### PR TITLE
Detect rust panic

### PR DESCRIPTION
**Pull request by Avgor46**

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 use crate::execution_class::ExecutionClass;
 use regex::Regex;
 
-/// Extract C++ exception info from stderr
+/// Extract exception info from stderr
```
Comment (SweetVishnya) [inline src/util.rs:4]: comment that cpp and rust are supported

File: src/util.rs [hunk 2]
```diff
@@ -34,13 +35,26 @@ pub fn cpp_exception_from_stderr(stderr_list: &[String]) -> Option<ExecutionClas
...
             instance,
             message,
             "",
-        )))
-    } else {
-        None
+        )));
     }
+    // Rust panic check
+    let rexception = Regex::new(r"thread '.+?' panicked at '(.+)?'").unwrap();
```
Comment (SweetVishnya) [inline src/util.rs:46]: We need some tests

File: src/util.rs [hunk 3]
```diff
@@ -34,13 +35,26 @@ pub fn cpp_exception_from_stderr(stderr_list: &[String]) -> Option<ExecutionClas
...
+        .iter()
+        .position(|line| rexception.is_match(line))
+    {
+        let message = rexception
+            .captures(&stderr_list[pos])
+            .unwrap()
+            .get(1)
+            .unwrap()
+            .as_str();
+        return Some(ExecutionClass::new(("NOT_EXPLOITABLE", "", message, "")));
```
Comment (SweetVishnya) [inline src/util.rs:57]: RustPanic for short description?

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: src/util.rs (hunk 1 above)
Comment (Avgor46) [reply src/util.rs]: done

File: src/util.rs (hunk 3 above)
Comment (Avgor46) [reply src/util.rs]: done

File: src/util.rs (hunk 2 above)
Comment (Avgor46) [reply src/util.rs:46]: Added test with panic in fuzz target

File: tests/casr_tests/test_rust_panic/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,3 @@
+fn main() {
+    println!("Hello, world!");
```
Comment (SweetVishnya) [inline tests/casr_tests/test_rust_panic/src/main.rs:2]: seems that there is no panic in this test

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: tests/casr_tests/test_rust_panic/src/main.rs (hunk 4 above)
Comment (Avgor46) [reply tests/casr_tests/test_rust_panic/src/main.rs:2]: test_rust_panic has a fuzz target with panic

**Review by SweetVishnya — APPROVED** [review]

**Pull request merged**